qirlib/src/context.rs

```diff
@@ -43,32 +46,36 @@ pub struct Context<'ctx> {
     pub constants: Constants<'ctx>,
 }
 
-pub enum ContextType<'ctx> {
+#[derive(Clone, Copy)]
+pub enum ModuleType<'ctx> {
```

Comment:
I think that this name is more accurate than `ContextType`, but I'm not sure it's quite right either. If you load a module from a template, a file, or (in the future) a byte array, the resulting module is the same type in all cases - the only difference is where it came from. Maybe it's a `ModuleLoader` or something like that?

Reply:
I agree.
- `ModuleLoader`
- `ModuleSource`
- `ModuleType`
- ?

Reply:
I'd be ok with `ModuleSource` or `ModuleLoader`. I can't think of any other names right now.